### ISSUE TITLE
Issues creating tables: panic: assignment to entry in nil map

### ISSUE THREAD

**Issue by textbook** [terraform-providers/terraform-provider-aws#8214]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
2019/04/05 12:38:38 [INFO] Terraform version: 0.11.8  7a5c1d221ac209bbac66947c369815cd9ca70ed5
2019/04/05 12:38:38 [INFO] Go runtime version: go1.10.1
```

(Also tried with 0.11.13, same behaviour)

### Affected Resource(s)

* aws_v2.5.0, `aws_dynamodb_table`

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "eu-west-2"
  version = "~> 2.5.0"

  endpoints {
    dynamodb = "http://dynamo:8000"
  }
}

resource "aws_dynamodb_table" "logins-dynamodb-table" {
  name           = "logins"
  hash_key       = "username"
  read_capacity  = "1"
  write_capacity = "1"

  attribute {
    name = "username"
    type = "S"
  }
}
```

### Debug Output

https://gist.github.com/textbook/74ae81de86d1a4d9f19a4c89ff637d2c

### Panic Output

```
2019-04-05T12:38:39.071Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 2019/04/05 12:38:39 [INFO] No assume_role block read from configuration
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: panic: assignment to entry in nil map
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: goroutine 34 [running]:
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.providerConfigure(0xc000954770, 0x0, 0xc00063a6e0, 0xc000954770, 0x0)
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/provider.go:1010 +0xda9
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Configure(0xc0009544d0, 0xc00094b470, 0x7f249d3f6000, 0x0)
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:264 +0xcd
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Configure(0xc000129ee0, 0xc00094b470, 0xc000ae81f8, 0x0, 0x0)
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:517 +0x3e
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: reflect.Value.call(0xc00011f1a0, 0xc000ae8008, 0x13, 0x3cd8565, 0x4, 0xc0009faf18, 0x3, 0x3, 0xc000127ac0, 0x413257, ...)
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/goenv/versions/1.11.5/src/reflect/value.go:447 +0x454
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: reflect.Value.Call(0xc00011f1a0, 0xc000ae8008, 0x13, 0xc00038d718, 0x3, 0x3, 0x12a05f200, 0xc00038d710, 0xc00038d7b8)
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/goenv/versions/1.11.5/src/reflect/value.go:308 +0xa4
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: net/rpc.(*service).call(0xc00007a0c0, 0xc000ab8000, 0xc00031c060, 0xc00031c070, 0xc00010a680, 0xc00063a020, 0x3a53d60, 0xc00094b470, 0x16, 0x2f40820, ...)
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/goenv/versions/1.11.5/src/net/rpc/server.go:384 +0x14e
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: created by net/rpc.(*Server).ServeCodec
2019-04-05T12:38:39.082Z [DEBUG] plugin.terraform-provider-aws_v2.5.0_x4: 	/opt/goenv/versions/1.11.5/src/net/rpc/server.go:481 +0x47e
```

### Expected Behavior

Table gets created as required (and as works fine in 2.4.0).

### Steps to Reproduce

1. `terraform apply -auto-approve`

### Important Factoids

Running this against the `amazon/dynamodb-local` container image in a network defined by Docker Compose.

**Comment by dreamlibrarian** [comment]
Amplifying this: any endpoint customization blows up under 2.5.0, because we're not initializing the new Endpoints map before using it.

Probably just needs a make(map[string]string) in the Config initialization block? 

https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/provider.go#L1010

**Comment by bflad** [comment]
Hi Folks 👋 Apologies for the unexpected behavior with the last release. I have submitted a fix with a covering test here, which we will release early next week: https://github.com/terraform-providers/terraform-provider-aws/pull/8226

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.6.0 of the Terraform AWS Provider later today 👍

**Comment by bflad** [comment]
The fix for this has been released in [version 2.6.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#260-april-10-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
